Frameworks/UIKit/UIAlertView.mm

```diff
@@ -1,641 +1,671 @@
+    [ret setBackgroundImage:buttonPressed forState:1];
+    [ret setTag:index];
+    [ret addTarget:self action:@selector(buttonClicked:) forControlEvents:UIControlEventTouchUpInside];
+    [ret setAlpha:0.9f];
+
+    return ret;
+}
+
+- (id)_didHideAlert {
+    [alertPriv->_darkView removeFromSuperview];
+    alertPriv->_darkView = nil;
+    [self removeFromSuperview];
+    alertPriv->_isVisible = false;
+
+    id weakDelegate = objc_loadWeak(&alertPriv->_delegate);
+    if (alertPriv->_delegateSupportsDidDismiss) {
+        [weakDelegate alertView:self didDismissWithButtonIndex:alertPriv->_hideIndex];
+    }
+    return self;
+}
+
+static void hideAlert(UIAlertView* self, int index, BOOL animated) {
+    if (animated) {
+        self->alertPriv->_hideIndex = index;
+        [UIView beginAnimations:@"HideAlert" context:nil];
+        [UIView setAnimationDuration:0.25f];
+        [UIView setAnimationDelegate:self];
+        [UIView setAnimationDidStopSelector:@selector(_didHideAlert)];
+        [self->alertPriv->_darkView setAlpha:0.0f];
+        [self setAlpha:0.0f];
+        [UIView commitAnimations];
+    } else {
+        [self->alertPriv->_darkView removeFromSuperview];
+        self->alertPriv->_darkView = nil;
+        [self removeFromSuperview];
+        self->alertPriv->_isVisible = false;
+
+        id weakDelegate = objc_loadWeak(&self->alertPriv->_delegate);
+        if ([weakDelegate respondsToSelector:@selector(alertView:didDismissWithButtonIndex:)]) {
+            [weakDelegate alertView:self didDismissWithButtonIndex:index];
+        }
+    }
+}
+
+/**
+ @Status Interoperable
+*/
+- (void)dismissWithClickedButtonIndex:(int)index animated:(BOOL)animated {
+    [self retain];
+    [self autorelease];
+
+    id weakDelegate = objc_loadWeak(&alertPriv->_delegate);
+
+    if ([weakDelegate respondsToSelector:@selector(alertView:willDismissWithButtonIndex:)]) {
+        [weakDelegate alertView:self willDismissWithButtonIndex:index];
+    }
+
+    hideAlert(self, index, animated);
+}
+
+- (void)buttonClicked:(id)button {
+    int index = [(UIButton*)button tag];
+
+    [self retain];
+    [self autorelease];
+
+    id weakDelegate = objc_loadWeak(&alertPriv->_delegate);
+
+    if ([weakDelegate respondsToSelector:@selector(alertView:clickedButtonAtIndex:)]) {
+        [weakDelegate alertView:self clickedButtonAtIndex:index];
+    }
+
+    if ([weakDelegate respondsToSelector:@selector(alertView:willDismissWithButtonIndex:)]) {
+        [weakDelegate alertView:self willDismissWithButtonIndex:index];
+    }
+
+    hideAlert(self, index, TRUE);
+}
+
+/**
+ @Status Interoperable
+*/
+- (BOOL)isVisible {
+    return alertPriv->_isVisible;
+}
+
+- (void) _handleRotation {
+    // We need to offset the 'presentationTransform' used in WOCDispalyMode, because we always want to render alerts vertically.
+    // TODO: We'll remove this when we switch UIAlertView over to a Xaml ContentDialog, since rotation will be handled for us.

```

Comment:
Basically this whole file will be gutted at that time.  I don't think this particular backlog item has been filed on github though...